lib/webResource.ts

```diff
@@ -271,10 +286,11 @@ export class WebResource {
 export interface RequestPrepareOptions {
   method: HttpMethods;
   url?: string;
-  queryParameters?: { [key: string]: any } | ParameterValue;
+  queryParameters?: { [key: string]: string | ParameterValue };
```

Comment:
How about this `string | number | boolean | object | array | ParameterValue`?

Reply:
This would still make sure that null, undefined will not get in. However it will let the user know what he/she can provide.

Reply:
Assuming we deal with things missing a `value` property by just calling `.toString()` on them (maybe implicitly through a template string), I think I'd be happiest with `{ toString(): string }` as the type. This allows `number`, `boolean`, `object`, array, etc.

`array` is not a type so it would have to be either any[] or something recursive.

Reply:
Talked it over a bit more and concluded that `any | ParameterValue` makes the most sense for now